### ISSUE TITLE
Way Too Loud

### ISSUE THREAD

**Issue by Katnami-Kode** [aqariio/Fowl-Play#22]
More birds = more noise. Since many birds spawn in groups, this is a major problem. Mod needs some kind of system to limit the amount of bird sounds going off at once. Additionally no fade in or fade out based on player distance - if there is a bird nearby, you're going to hear it at full blast. 
There is no way to edit the volume of birds in the in-game config. 

I like this mod - the models and animations are smooth and cute - but I can't use it in its current, noisy state.

**Comment by aqariio** [comment]
There is fade out, or attenuation, for bird sounds, although their attenuation distance is much greater than vanilla mobs. This way, players can hear birds from further away. 

A sound attenuation option in the config (for every bird) is planned for the next update. I don't plan on adding a volume option at this time, but you can use a mod that allows customization of individual sounds, like https://modrinth.com/mod/sound-controller. 

If you have the time, could you send a video (with sound) of the birds being loud? I've received a few complaints about this, but it has always confused me, as I don't find the birds loud at all.

**Comment by GitCodMaster** [comment]
Can't we just get an update where we can set the max volume of every bird type/group?

**Comment by aqariio** [comment]
Again, if you want an immediate solution, use a mod that allows you to customize the volumes of individual sound events, like https://modrinth.com/mod/sound-controller. 

I will add an option in the future for individual volume customization, but it's not something I can just implement in a day or two.

> i also noticed this, its extremely loud and my master volume is at 50%
> ive come across rivers with a huge amount of ducks as well and you could hear all of them, which again, very loud


Could you send a video (with sound) of the birds being loud?

**Comment by Katnami-Kode** [comment]
> If you have the time, could you send a video (with sound) of the birds being loud? I've received a few complaints about this, but it has always confused me, as I don't find the birds loud at all.

mm. The volume of birdcalls isn't louder than that of other passive mobs - _however,_ because of the increased attenuation distance, players are much more likely to hear every bird in their area at maximum volume. This problem scales with the number of birds too, as they're all competing for the same volume range, it ends up sounding very cluttered (loud).
While I think the attenuation distance should be greater than of other passive mobs, I also think the maximum volume needs to be adjusted. 

Birdsong is, for the most part, an ambient effect, while passive mob noises are there to give life to specific actors in the world. While birdsong is tied to specific actors, it's meant to provide information about the environment, and not the entities within the environment. I don't really need to know that there is a robin in my area - if I listen actively I might be able to recognize that, but the more important aspect is the atmosphere created - It lets me know that this biome is vibrant and alive. 
Ambient effects are best mixed lower, so they don't disrupt or overshadow more immediately important information (such as a zombie growling, or a block breaking). Along with that, since ambient effects are heard more frequently, having them at a high volume can make them get grating or repetitive. 

TLDR: because bird sounds are heard more frequently, and because their presence is less immediately important than that of other mobs, they should be quieter.

**Comment by aqariio** [comment]
I think you have a good point. In the future I would like to attenuate their calls on a curve, so that their sounds soften much more quickly as distance grows. I'll focus on regulating their volumes and attenuation distances in the next update, and hopefully that will make them much less distracting.

**Comment by Katnami-Kode** [comment]
Ah, thank you ^^
A steeper attenuation curve sounds like a good idea.

**Comment by Avamander** [comment]
I just noticed the same (on the latest available version for 1.21.1) and one of the reasons might be that birds tend to make really high-frequency sounds and chirps, so due to that they might need some additional attenuation (to sound less abrasive) compared to other sounds at a similar volume level.
